src/compiler/transformers/declarations.ts

```diff
@@ -1006,7 +1006,9 @@ namespace ts {
                             if (!isPropertyAccessExpression(p.valueDeclaration)) {
                                 return undefined;
                             }
+                            getSymbolAccessibilityDiagnostic = createGetSymbolAccessibilityDiagnosticForNode(p.valueDeclaration);
                             const type = resolver.createTypeOfDeclaration(p.valueDeclaration, enclosingDeclaration, declarationEmitNodeBuilderFlags, symbolTracker);
+                            getSymbolAccessibilityDiagnostic = oldDiag;

```

Comment:
I don't know this part of the codebase, but I don't get why you're not stashing into `oldDiag` first.

Reply:
Because we already stashed it for the context we made for the associated function declaration ❤️